InnoClinic.OfficesAPI/UnitTests/OfficeServiceTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Text;
+using AutoMapper;
+using BLL.DTO;
+using BLL.Services;
+using DAL.Entities;
+using DAL.Interfaces;
+using FluentAssertions;
+using Microsoft.AspNetCore.Http;
+using MongoDB.Bson;
+using MongoDB.Driver;
+using Moq;
+using UnitTests.TestCases;
+
+namespace UnitTests
+{
+    public class OfficeServiceTests
+    {
+        private readonly Mock<IOfficeRepository> _officeRepositoryMock;
+        private readonly Mock<IPhotoRepository> _photoRepositoryMock;
+        private readonly Mock<IMapper> _mapperMock;
+        private readonly OfficeService _officeService;
+
+        public OfficeServiceTests()
+        {
+            _officeRepositoryMock = new Mock<IOfficeRepository>();
+            _photoRepositoryMock = new Mock<IPhotoRepository>();
+            _mapperMock = new Mock<IMapper>();
+            _officeService = new OfficeService(_officeRepositoryMock.Object, _photoRepositoryMock.Object, _mapperMock.Object);
+        }
+
+        [Fact]
+        public async Task CreateOfficeAsyncReturnsOfficeResultDto()
+        {
+            var dto = new OfficeRequestDto
+            {
+                City = TestConstants.TestCity,
+                Street = TestConstants.TestStreet,
+                HouseNumber = TestConstants.TestHouseNumber,
+                OfficeNumber = TestConstants.TestOfficeNumber,
+                Photo = CreateTestFormFile()
+            };
+
+            var office = new Office();
+            var createdOffice = new Office { Id = TestConstants.TestOfficeId };
+            var expectedResult = new OfficeResultDto();
+
+            _mapperMock.Setup(m => m.Map<Office>(dto)).Returns(office);
+            _officeRepositoryMock.Setup(r => r.CreateOfficeAsync(office, It.IsAny<CancellationToken>())).ReturnsAsync(createdOffice);
+            _mapperMock.Setup(m => m.Map<OfficeResultDto>(createdOffice)).Returns(expectedResult);
+            _photoRepositoryMock.Setup(r => r.UploadPhotoAsync(It.IsAny<string>(), It.IsAny<byte[]>(), It.IsAny<CancellationToken>())).ReturnsAsync(TestConstants.TestPhotoFileId);
+
+            var result = await _officeService.CreateOfficeAsync(dto);
+
+            result.Should().Be(expectedResult);
```

Comment:
Can you also check here if the repo's methods have been called and how many times?

Reply:
And do it for other test methods as well